### PR TITLE
[Enhancement] Use batch delete in cloud native vacuum

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 27 / 29 (93.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|27|29|93.10%|[184, 192]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 27 / 29 (93.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|27|29|93.10%|[184, 192]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 46 / 48 (95.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|46|48|95.83%|[70, 379]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 46 / 48 (95.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|46|48|95.83%|[70, 379]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|43|46|93.48%|[66, 77, 369]|

File: be/src/storage/lake/vacuum.cpp [hunk 1]
```diff
@@ -60,16 +60,23 @@ static Status delete_file(FileSystem* fs, const std::string& path) {
...
-        auto seg_path = join_path(data_dir, segment);
-        RETURN_IF_ERROR(delete_file_ignore_not_found(fs, seg_path));
+static Status delete_files(FileSystem* fs, const std::vector<std::string>& paths) {
+    auto wait_duration = config::experimental_lake_wait_per_delete_ms;
+    if (wait_duration > 0) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(wait_duration));
     }
-    return Status::OK();
+    for (auto&& path : paths) {
+        LOG_IF(INFO, config::lake_print_delete_log) << "Deleted " << path;
```
Comment (sduzh) [inline be/src/storage/lake/vacuum.cpp:69]:
```suggestion
        LOG_IF(INFO, config::lake_print_delete_log) << "Deleting " << path;
```

File: be/src/storage/lake/vacuum.cpp [hunk 2]
```diff
@@ -129,16 +136,20 @@ static Status vacuum_tablet_metadata(TabletManager* tablet_mgr, std::string_view
...
-                    RETURN_IF_ERROR(delete_rowset_files(fs.get(), data_dir, rowset));
+                    for (const auto& segment : rowset.segments()) {
+                        paths.emplace_back(std::move(join_path(data_dir, segment)));
+                    }
                     *vacuumed_files += rowset.segments_size();
                     *vacuumed_file_size += rowset.data_size();
                 }
                 for (const auto& file : metadata->orphan_files()) {
-                    RETURN_IF_ERROR(delete_file_ignore_not_found(fs.get(), join_path(data_dir, file.name())));
+                    paths.emplace_back(std::move(join_path(data_dir, file.name())));
```
Comment (sduzh) [inline be/src/storage/lake/vacuum.cpp:148]:
```suggestion
                    paths.emplace_back(join_path(data_dir, file.name()));
```

File: be/src/storage/lake/vacuum.cpp [hunk 3]
```diff
@@ -129,16 +136,20 @@ static Status vacuum_tablet_metadata(TabletManager* tablet_mgr, std::string_view
                 return res.status();
             } else {
                 auto metadata = std::move(res).value();
+                std::vector<std::string> paths;
                 for (const auto& rowset : metadata->compaction_inputs()) {
-                    RETURN_IF_ERROR(delete_rowset_files(fs.get(), data_dir, rowset));
+                    for (const auto& segment : rowset.segments()) {
+                        paths.emplace_back(std::move(join_path(data_dir, segment)));
```
Comment (sduzh) [inline be/src/storage/lake/vacuum.cpp:142]:
```suggestion
                        paths.emplace_back(oin_path(data_dir, segment));
```

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 40 / 43 (93.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|40|43|93.02%|[66, 77, 369]|

File: be/src/storage/lake/vacuum.cpp [hunk 4]
```diff
@@ -60,16 +60,23 @@ static Status delete_file(FileSystem* fs, const std::string& path) {
...
+    if (wait_duration > 0) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(wait_duration));
     }
-    return Status::OK();
+    for (auto&& path : paths) {
+        LOG_IF(INFO, config::lake_print_delete_log) << "Deleting " << path;
+    }
+    auto t0 = butil::gettimeofday_us();
+    auto st = fs->delete_files(paths);
+    if (st.ok()) {
```
Comment (sduzh) [inline be/src/storage/lake/vacuum.cpp:73]:
```suggestion
    if (st.ok()) {
        LOG(INFO, config::lake_print_delete_log) << "Deleted " << paths.size() << " files";
```

Comment (TszKitLo40) [reply be/src/storage/lake/vacuum.cpp:73]: done

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 40 / 43 (93.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|40|43|93.02%|[66, 77, 369]|

File: be/src/storage/lake/vacuum.cpp (hunk 4 above)
Comment (sduzh) [reply be/src/storage/lake/vacuum.cpp:73]: Should be `LOG_IF` not `LOG`

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 41 / 44 (93.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|41|44|93.18%|[66, 78, 370]|

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**